### ISSUE TITLE
ReactiveBase tries to parse all query parameters as JSON

### ISSUE THREAD

**Issue by thebuilder** [appbaseio/reactivesearch#607]
**Issue Type:**
bug

**Platform:**
web

**Description:**
When creating a new `<ReactiveBase>` it tries to parse all the query parameters as JSON strings.
If it encounters any parameter that fails to parse as JSON, it will stop and log an error to the console.

```
index.js:1 REACTIVESEARCH - An error occured while parsing the URL state. SyntaxError: Unexpected token M in JSON at position 0
    at JSON.parse (<anonymous>)
    at ReactiveBase.js:147
    at Array.forEach (<anonymous>)
    at ProxyComponent.setStore (ReactiveBase.js:144)
    at new ReactiveBase (ReactiveBase.js:67)
```

https://github.com/appbaseio/reactivesearch/blob/7a9fb9bd39213bf4fde0f5268d4c9350ac4bedaa/packages/web/src/components/basic/ReactiveBase.js#L87

- When rendering the `<ReactiveBase>` inside a Storybook (that uses query paramters) it will always log an error.
- If the application has it's own query parameters (that are not JSON), it will not get any values.

**Anything else:**
Remove the `console.error` message - It's not helpful to print the stack trace when a json parse fails.

If it is necessary for the query parameters you use to be JSON encoded, i'd recommend a check before doing `JSON.parse`. Does the value start with `"` could be a simple check. Maybe fallback to the plain value if it isn't a JSON string? So you could do `?query=searchterm`.

Adding the `try...catch` around each parse could also make sense, so even if a parse fails, it will still find the remaining valid values

The option to disable it would also make sense, since you can disable URLParams or search components.

**Comment by calebdel** [comment]
I also encountered this today.

**Comment by calebdel** [comment]
Same issue today. As suggested this solution seems to work:
`              params.get(key).charAt(0) == '"'
                ? params.get(key)
                : JSON.parse(params.get(key))`

**Comment by redlinecodes** [comment]
@metagrover - I've been having this issue for awhile and just saw this. If I update to the latest version will that fix query parameters issue?

**Comment by metagrover** [comment]
This issue hasn't been fixed yet. It will be released in `v3.0.0`.

**Comment by siddharthlatest** [comment]
@metagrover Since this should ideally be a backward compatible change (right?), we can also patch it in the next `2.x.y` release.

**Comment by metagrover** [comment]
Yes, makes sense.

**Comment by rap2hpoutre** [comment]
@thebuilder @calebdel @redlinecodes Fixed via #752 & #754, available in v2.16.1.

Thanks to @metagrover for the quick release!
